update/Pos/Update/Poll/PollState.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- Types related Poll update system
+
+module Pos.Update.Poll.PollState
+       (
+         -- * Types
+         PollState (..)
+
+       -- * Lenses
+       -- * PollState
+       , psBlockVersions
+       , psAdoptedBV
+       , psEpochProposers
+       , psConfirmedBVs
+       , psConfirmedProposals
+       , psActiveProposals
+       , psDelActivePropsIdx
+       , psSlottingData
+       , psFullRichmenData
+       , psIssuersStakes
+
+       -- * Functions
+       , modifyPollState
+       ) where
+
+import           Universum
+
+import           Control.Lens                 (makeLenses)
+import           Data.HashSet                 (HashSet)
+import qualified Data.HashMap.Strict          as HM
+
+import           Pos.Core.Types               (ApplicationName, BlockVersion,
+                                               BlockVersionData, EpochIndex,
+                                               NumSoftwareVersion, SoftwareVersion,
+                                               StakeholderId)
+import           Pos.Lrc.DB.Issuers           (IssuersStakes)
+import           Pos.Lrc.Types                (FullRichmenData)
+import           Pos.Slotting.Types           (SlottingData)
+import           Pos.Update.Core              (UpId)
+import           Pos.Update.Poll.Types        (BlockVersionState, ConfirmedProposalState,
+                                               PollModifier (..), ProposalState)
+import           Pos.Util.Modifier            (modifyHashMap)
+
+data PollState = PollState
+    { -- | All competing block versions with their states
+      _psBlockVersions      :: !(HashMap BlockVersion BlockVersionState)
+      -- | Presently adopted block version with its data
+    , _psAdoptedBV          :: !(BlockVersion, BlockVersionData)
+      -- | All stakeholders who made proposals in the current epoch
+    , _psEpochProposers     :: !(HashSet StakeholderId)
+      -- | All applications in use and their latest (confirmed) versions
+    , _psConfirmedBVs       :: !(HM.HashMap ApplicationName NumSoftwareVersion)
+      -- | All proposed software versions and their state
+    , _psConfirmedProposals :: !(HM.HashMap SoftwareVersion ConfirmedProposalState)
+      -- | All update proposals and their states
+    , _psActiveProposals    :: !(HM.HashMap UpId ProposalState)
+      -- | Update proposals for each application
+    , _psDelActivePropsIdx  :: !(HM.HashMap ApplicationName (HashSet UpId))
```

Comment:
Btw, in DB we don't store this index anymore (we iterate through proposals and filter by name), but I think it's even good two have different implementation of the same behaviour so that they can verify each other.